src/transport_picoquic.cpp

```diff
@@ -596,14 +596,10 @@ PicoQuicTransport::start()
 
 void PicoQuicTransport::pq_runner() {
 
-    while (picoquic_runner_queue.size() > 0) {
-        auto cb = std::move(picoquic_runner_queue.pop());
-        if (cb) {
+    do {
+        if(auto cb = std::move(picoquic_runner_queue.pop()))
             (*cb)();
-        } else {
-            break;
-        }
-    }
+    } while (picoquic_runner_queue.size() > 0);
```

Comment:
My second comment may have been more sane, I don't know why do/while came to mind first. This also just occurred to me as another valid way of doing this with fewer branches (and actually, may be what the compiler does under the hood, who knows):
```cpp
while (auto cb = std::move(picoquic_runner_queue.pop())) {
    (*cb)();
}

Reply:
Is the move causing performance issues? Is it hiding copy elision?

Reply:
Doesn't seem to make a diff on perf, but it's nicer to reduce lines.